kuzzleio/kuzzle
# Customize Open API specification in plugins/controllers

Closes #2122 

## What does this PR do ?

With this PR, a new property `openapi` can be use while extending your app with new actions. You can now provide your own custom OpenAPI specification v3 when the one auto-generated by Kuzzle is not adapted.

### How should this be manually tested?

 - Clone Kuzzle
 - Add a controller with openAPI specification as detailed in the doc
 - Run Kuzzle
 - Request `server:openapi`, the result should contain the same openAPI documentation as given for those actions

### What's next?

 - Custom specification provided by this feature may provide enough information to set up a validation for these routes (currently, it is handled by the plugin developer)
 - OpenAPI specification top fields could be also customizable (it is just hard coded as from now)
